### PR TITLE
Nox stuff

### PR DESCRIPTION
**Pull request by Endercheif**
### Changes

It has come to my attention that many of the measure we have in place are not being run or used as much. Nox will fix that. 
Before doing a pr, on can run `nox` on their local machine to run tests and lints. As a workflow, they will also run.

 ### Check off the following

-   [x] I have tested my changes with the current requirements
-   [x] My Code follows the pep8 code style.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -74,5 +75,5 @@ warn_unused_configs = true
 no_implicit_reexport = true
 
 [tool.black]
-line-length = 80
+line-length =  88
```
Comment (Arthurdw) [inline pyproject.toml:78]: Any reason for this change?

Comment (Endercheif) [reply pyproject.toml:78]: In .flake8 and .editorconfig it is set to 88

**Comment by Sigmanificient** [comment]
What is tox doing?

**Comment by Endercheif** [comment]
> What is tox doing?

tox (thing I removed) did nothing because it was never integrated into the code base, it was an extra dependancy. nox (what I added) runs black, isort, flake8, pytest, mypy all in one command. Prior to this, it was clear many had not ran these individually as they were supposed to. nox makes it easy and will also auto run as a GitHub workflow

**Comment by Sigmanificient** [comment]
> > What is tox doing?
> 
> tox (thing I removed) did nothing because it was never integrated into the code base, it was an extra dependancy. nox (what I added) runs black, isort, flake8, pytest, mypy all in one command. Prior to this, it was clear many had not ran these individually as they were supposed to. nox makes it easy and will also auto run as a GitHub workflow

We maybe dont want to run mypy cause there is a lot of type errors currently

**Review by trag1c — APPROVED** [review]

**Review by Sigmanificient — APPROVED** [review]

**Comment by Sigmanificient** [comment]
nox still failling xD

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      matrix:
+        os: [ ubuntu, windows, macos ]
+
+    name: Nox on ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Python Setup
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
```
Comment (Lunarmagpie) [inline .github/workflows/ci.yml:19]: Nox should be run on all supported versions of python. You should use the matrix thing. Heres an example from one of my projects https://github.com/magpie-dev/hikari-crescent/blob/main/.github/workflows/pytest.yml

File: noxfile.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+@nox.session
+def lint(session: nox.Session):
+    session.install("flake8")
+    session.run("flake8", "pincer")
+
+
+@nox.session
+def mypy(session: nox.Session):
+    session.install("mypy")
+    session.run("mypy", "pincer")
```
Comment (Lunarmagpie) [inline noxfile.py:27]: Mypy shouldn't be run. Theres far too many type issues so its always gonna fail.

**Review by Lunarmagpie — CHANGES_REQUESTED** [review]

File: noxfile.py (hunk 3 above)
Comment (Sigmanificient) [reply noxfile.py:27]: Yes that what i was saying too, we should run it later

Comment (Sigmanificient) [inline noxfile.py:27]:
```suggestion
"""
Mypy will be be ran within a workflow until pincer provide complete typing
that will not fail for every push.

@nox.session
def mypy(session: nox.Session):
   session.install("mypy")
   session.run("mypy", "pincer")
"""
```

File: pyproject.toml [hunk 4]
```diff
@@ -25,7 +25,7 @@ classifiers = [
 "Discord" = "https://discord.gg/8WkYz3fNFm"
 
 [tool.poetry.dependencies]
-python = ">=3.8"
+python = ">=3.8,<3.11"
```
Comment (Sigmanificient) [inline pyproject.toml:28]: Breaking changes on python 3.11?

**Review by Sigmanificient — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 4 above)
Comment (Arthurdw) [reply pyproject.toml:28]: I don't think there are breaking changes for us.

Comment (Sigmanificient) [reply pyproject.toml:28]: Why changing this then?

File: .github/workflows/ci.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (Sigmanificient) [reply .github/workflows/ci.yml:19]: @Endercheif

File: noxfile.py (hunk 3 above)
Comment (Sigmanificient) [reply noxfile.py:27]: @Endercheif

File: pyproject.toml (hunk 4 above)
Comment (Endercheif) [reply pyproject.toml:28]: @Sigmanificient @Arthurdw Poetry is very picky about supported version of dependencies. One such dependency (Nox I believe) does not support anything greater than 3.11 so the install fails. Unless this is added.

**Comment by Sigmanificient** [comment]
Can we just format the coe with black one and for all?

**Comment by Endercheif** [comment]
The only reason the Nox is failing, is because the code is not formatted correctly. 
We have two options. 
- Merge this now and in a different PR resolve the formatting issue or 
- In this PR, format it.

I would say the second one just to keep the PRs organized.

**Review by Sigmanificient — APPROVED** [review]

**Review by Arthurdw — APPROVED** [review]
Seems okay to me.

**Pull request merged**